Common/AssemblyResolver.cs

```diff
@@ -75,28 +91,28 @@ public AssemblyResolver(ILogger logger, params string[] rootSearchPaths)
         }
 
         /// <summary>
-        /// Attempts to create the name of the file associated with a given assembly name.
+        /// Turns the input assembly ref argument into an files name.
+        /// The input might be a file name (e.g. foo.dll) or a full assembly name
+        /// (e.g. SimpleAssemblyByFullName, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null)
         /// </summary>
-        public static string CreateFileNameFromAssemblyName(string input) // Public for testing purposes
+        private static string GetAssemblyFileName(string input)
         {
             Debug.Assert(input != null);
             Debug.Assert(input.Length > 0);
 
-            if (input.EndsWith(".dll"))
+            if (IsFileName(input))
             {
                 return input;
             }
 
-            string result = input;
-            if (input.Contains(" "))
-            {
-                // If the assembly name has multiple words (seperated by spaces), use only the first word
-                // (e.g. "foo bar" -> "foo")
-                string[] parts = input.Split(new char[] { ' ' });
-                result = parts[0].Substring(0, parts[0].Length - 1);
-            }
+            AssemblyName assemblyName = new AssemblyName(input);
+            return assemblyName.Name + DllExtension;
+        }
 
-            return  result + ".dll";
+        private static bool IsFileName(string fileName)
```

Comment:
IsAssemblyByFileName ? Since there are other areas where this logic happens, consider extracting to an Utils class.